src/Microsoft.Azure.CosmosEventSourcing/Stores/IEventStore.cs

```diff
@@ -23,6 +23,37 @@ public interface IEventStore<TEventItem> where TEventItem : EventItem
         IEnumerable<TEventItem> items,
         CancellationToken cancellationToken = default);
 
+    /// <summary>
+    /// Persists a set of <see cref="EventItem"/> items.
+    /// </summary>
+    /// <param name="aggregateRoot">The aggregate containing the events to persist.</param>
+    /// <param name="cancellationToken">A token that can be used to cancel this async request.</param>
+    ValueTask PersistAsync(
+        IAggregateRoot aggregateRoot,
+        CancellationToken cancellationToken = default);
+
+    /// <summary>
+    /// Persists a set of <see cref="EventItem"/> items.
+    /// </summary>
+    /// <param name="aggregateRoot">The aggregate containing the events to persist.</param>
+    /// <param name="partitionKeyValue">The partition key value to use.</param>
+    /// <param name="cancellationToken">A token that can be used to cancel this async request.</param>
```

Comment:
Might be worth add some `<remarks></remarks>` to this just stating it is using `Activator`